### ISSUE TITLE
Homkit shows IEEE address and not friendly name

### ISSUE THREAD

**Issue by Tonitom** [itavero/homebridge-z2m#473]
When I add a device in z2m, then the IEEE address is displayed in Homebridge as the friendly name, when I change the name in z2m, the address name remains. That means before adding a zigbee device in z2m I have to turn off homebridge and after renaming I can start again.

Is there a trick?

**Comment by itavero** [comment]
Please see #76 as well.

If I remember correctly, simply restarting Homebridge after the rename used to do the trick. As usual, a restart of the Home.app might also be needed.
Does this work for you as well?

**Comment by Tonitom** [comment]
I deleted the device in z2m and reconnected it. Restart doesn't work.


Just some ideas after I also tested the mqttthings plugin.
1. is it possible to disable automatic matching of the configuration? And ideally with a fake switch in Homekit for updating. 

2. and it would be great if the mapping is optional only by friendly name and not by IEEE address. For example, if a light bulb fails, I have to reconfigure everything in Homekit, if it's just by name, I only have to replace the bulb in z2m with the correct name.

**Comment by itavero** [comment]
> 1. is it possible to disable automatic matching of the configuration? And ideally with a fake switch in Homekit for updating.

You could exclude all devices by default and only include the devices you want to use, by manually setting `exclude` to `false` based on their friendly name or IEEE address. I believe there's an example of that in the [documentation](https://z2m.dev/config.html#defaults).

> 2. and it would be great if the mapping is optional only by friendly name and not by IEEE address. For example, if a light bulb fails, I have to reconfigure everything in Homekit, if it's just by name, I only have to replace the bulb in z2m with the correct name.

Can you rephrase this? I'm not sure what you are asking for here.

**Comment by Tonitom** [comment]
As I understand it, in HomeKit the Zigbee devices are created with friendly names and the IEEE address becomes the serial number.
When a Zigbee bulb is replaced, a new bulb is created in HomeKit and the old bulb is deleted. Room mapping and automations then need to be edited.

If the mapping is done by friendly name only, then the replacement should be easier as Homebridge ideally does not need to provision it as a new device and can keep the old one.

**Comment by Tonitom** [comment]
Possibly I am overthinking this. I come from Homematic and have already lost several actors and sensors. The devices that I could replace with the same type were automatically replaced in all automations and direct links, if I replaced the device with a newer generation, then the configuration was a pain. That's where my mindset comes from.

But I understand that a non-unique mapping can cause problems and together with Zigbee2MQTT and the MQTTThings plugin I have a reliable solution. 

thanks

**Comment by smulle48** [comment]
I am experiencing the same, but only for buttons/switches.

Bulbs, blinds, etc are named correctly even without restart.

Buttons don't update their name by restart. Sometimes they update the name after a few hours (cashing??) or they simply don't update at all

**Comment by itavero** [comment]
> Buttons don't update their name by restart. Sometimes they update the name after a few hours (cashing??) or they simply don't update at all

The Home.app might do caching, but these devices also have a quite specific service handler in this plugin. I can have a look if I can spot something in the plugin that might cause this.
